include/range/v3/detail/range_access.hpp

```diff
@@ -338,17 +338,14 @@ namespace ranges
             sized_sentinel_for_cursor<T, T> && //
             CPP_fragment(detail::random_access_cursor_, T);
         template<typename T>
-        CPP_concept_fragment(contiguous_cursor_,
-            requires(T & t) //
-            (
-                concepts::requires_<std::is_lvalue_reference<
-                    decltype(range_access::read(t))>::value>
-            ));
+        CPP_concept_bool contiguous_cursor_ = 
+        std::is_lvalue_reference<
+                    decltype(range_access::read(std::declval<T&>()))>::value;
```

Comment:
I think this is incorrect. You are in a non-SFINAE context here. If no `read` overload is found, this will be a hard error in concepts emulation mode.

What was wrong with the original?

Reply:
You're right.  MSVC `/std:c++latest` choked on the original version and I can't think of another workaround.

I opened a bug for MSVC: https://developercommunity.visualstudio.com/content/problem/1010462/concepts-requires-miscompilation.html.